portafly/src/components/data-list/reducers/TableReducer.ts

```diff
@@ -5,18 +5,41 @@ import {
   createReducer
 } from 'utils'
 import { DataListRow, DataListCol } from 'types'
+import { SortByDirection } from '@patternfly/react-table'
 
 export type TableState = {
   rows: DataListRow[]
   columns: DataListCol[]
+  sortBy?: {
+    index: number
+    direction: SortByDirection
+  }
 }
 
 // Action Handlers
+const SORT_BY = 'SORT_BY'
 const SELECT_ONE = 'SELECT_ONE'
 const SELECT_PAGE = 'SELECT_PAGE'
 const SELECT_ALL = 'SELECT_ALL'
 
 const tableActionHandlers: ActionHandlers<TableState, any> = {
+  [SORT_BY]: (state, action: Action<{ index: number, direction: SortByDirection }>) => {
+    const { index, direction } = action.payload
+    const sortedRows = state.rows.sort((a, b) => {
+      // When table is selectable, index must be corrected because of the first row of checkboxes
+      const prev = a.cells[index - 1]
+      const next = b.cells[index - 1]
+      if (prev < next) return -1
+      if (prev > next) return 1
+      return 0
+    })
+
+    if (direction === SortByDirection.desc) {
+      sortedRows.reverse()
+    }
+
+    return { ...state, rows: sortedRows, sortBy: action.payload }
+  },
```

Comment:
Careful, both `sort` and `reverse` change the original array, so here you are mutating the state. 
You could `slice` the original array before the sort, and that would be enough in this case. Also, better to export that logic into a separated function to make it clear the sorting, something like `sortByIndexCorrection` or `withIndexCorrection`

Reply:
> Careful, both `sort` and `reverse` change the original array, so here you are mutating the state.

👍 

> Also, better to export that logic into a separated function to make it clear the sorting, something like `sortByIndexCorrection` or `withIndexCorrection`

Now that I think about it, I don't suppose this belongs to the reducer. I'm taking it out and figure the _correction_ out besides the Table.